mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/reports/entry/ReportsEntryViewModel.kt

```diff
@@ -0,0 +1,466 @@
+package rs.readahead.washington.mobile.views.fragment.reports.entry
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.firebase.crashlytics.FirebaseCrashlytics
+import com.hzontal.tella_vault.VaultFile
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.Flowable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import rs.readahead.washington.mobile.MyApplication
+import rs.readahead.washington.mobile.data.database.DataSource
+import rs.readahead.washington.mobile.data.entity.reports.ReportBodyEntity
+import rs.readahead.washington.mobile.domain.entity.EntityStatus
+import rs.readahead.washington.mobile.domain.entity.UploadProgressInfo
+import rs.readahead.washington.mobile.domain.entity.collect.FormMediaFile
+import rs.readahead.washington.mobile.domain.entity.collect.FormMediaFileStatus
+import rs.readahead.washington.mobile.domain.entity.reports.ReportFormInstance
+import rs.readahead.washington.mobile.domain.entity.reports.TellaReportServer
+import rs.readahead.washington.mobile.domain.exception.NoConnectivityException
+import rs.readahead.washington.mobile.domain.repository.reports.ReportsRepository
+import rs.readahead.washington.mobile.domain.usecases.reports.*
+import rs.readahead.washington.mobile.util.fromJsonToObjectList
+import rs.readahead.washington.mobile.util.jobs.TellaRequestUploadJob
+import rs.readahead.washington.mobile.views.fragment.reports.adapter.ViewEntityTemplateItem
+import rs.readahead.washington.mobile.views.fragment.reports.mappers.toViewEntityInstanceItem
+import timber.log.Timber
+import javax.inject.Inject
+
+@HiltViewModel
+class ReportsEntryViewModel @Inject constructor(
```

Comment:
This is our only ViewModel regarding Reports so maybe we should remove Entry from the name.